pkg/controller/networkpolicy/validate.go

```diff
@@ -156,9 +249,21 @@ func NewNetworkPolicyValidator(networkPolicyController *NetworkPolicyController)
 	vr.RegisterTierValidator(&tv)
 	vr.RegisterGroupValidator(&gv)
 	vr.RegisterAdminNetworkPolicyValidator(&av)
+
+	// Set up Tier event handlers to notify validator when Tiers are actually created/deleted
+	networkPolicyController.SetupTierEventHandlersForValidator(&vr)
 	return &vr
 }
 
+// Run starts the background routines for the NetworkPolicyValidator.
+func (v *NetworkPolicyValidator) Run(stopCh <-chan struct{}) {
+	// Start cleanup routine for expired reservations from the tierValidator
+	for _, tv := range v.tierValidators {
+		go tv.startCleanupRoutine(stopCh)
```

Comment:
you call `Run` in a goroutine in `s.AddPostStartHook`, so you expect it to be blocking, so I wouldn't call `tv.startCleanupRoutine` in yet another goroutine

BTW, would it make sense to just replace `v.tierValidators` with a single `v.tierValidator`?